services/task/migrate.go

```diff
@@ -63,8 +64,7 @@ func runMigrateTask(ctx context.Context, t *admin_model.Task) (err error) {
 		t.EndTime = timeutil.TimeStampNow()
 		t.Status = structs.TaskStatusFailed
 		t.Message = err.Error()
-		// fixme: Because ctx is canceled here, so the db.DefaultContext is needed.
-		if err := t.UpdateCols(db.DefaultContext, "status", "message", "end_time"); err != nil {
```

Comment:
oops, this regression should not have occurred.
Thanks to @earl-warren for pointing it out.

Reply:
What did you mean?

Reply:
#27740 and #27741

Reply:
That is not a regression of this PR.

Reply:
It is, as this PR explicitly replaced it with a cancelable context again

Reply:
Ah no, sorry.
I misread the code.
You set `ctx = db.DefaultContext` above.